### PR TITLE
[CAZ-1488] Add Basic AUTH

### PR DESCRIPTION
**Pull request by likeuwill**
## Motivation and Context

[JIRA](https://eaflood.atlassian.net/browse/CAZ-1488)
## Checklist:

- [x] It contains only changes required by issue (does not contain other PR)
- [x] Includes link to a issue (if apply)
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: app/api_wrapper/api_exception.rb [hunk 1]
```diff
@@ -18,13 +18,13 @@ class ApiException < StandardError
   # ==== Attributes
   #
   # * +status+ - string or integer, HTTP error status
-  # * +status_message+ - string, HTTP error message
+  # * +message+ - string, HTTP error message
```
Comment (miqs1992) [inline app/api_wrapper/api_exception.rb]: Not sure about this change - there will be so many `message` in this class, instance variable, attr_reader, and a param

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -18,6 +18,8 @@ import { initAll } from 'govuk-frontend/govuk/all.js';
 import initBanner from '../src/cookiesBanner/init';
 import initPrintLink from '../src/printLink/init';
 
+document.body.classList.add('js-enabled');
```
Comment (miqs1992) [inline app/javascript/packs/application.js]: Is this removed from the view?

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -14,6 +14,13 @@ class ApplicationController < ActionController::Base
               BaseApi::Error400Exception,
               with: :redirect_to_server_unavailable
 
+  # enable basic HTTP authentication on production environment if HTTP_BASIC_PASSWORD variable present
+  http_basic_authenticate_with name: ENV['HTTP_BASIC_USER'],
```
Comment (miqs1992) [inline app/controllers/application_controller.rb]: Is this added to .env.example?

Comment (likeuwill) [reply app/controllers/application_controller.rb]: yes
```
# Basic Authentication
HTTP_BASIC_USER=
HTTP_BASIC_PASSWORD=
```

File: app/javascript/packs/application.js (hunk 2 above)
Comment (likeuwill) [reply app/javascript/packs/application.js]: you mean in app.htmk.haml this lines?
```
:javascript
      document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');
```
If yes, so the answer yes, removed.

**Pull request merged**